File: papers/src/main/java/com/pkdoc/papers/model/Paper.java
package com.pkdoc.papers.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import lombok.Data;

import java.time.LocalDate;
import java.util.Set;

@Entity
@Table(name = "PAPER")
@Data
public class Paper {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id")
    private Long id;

    @NotBlank
    @Column(name = "title")
    private String title;

    @Column(name = "abstract_text")
    private String abstractText;

    @Column(name = "authors")
    private String authors;

    @ManyToOne
    private User uploader;

    @Column(name = "publish_date")
    private LocalDate publishDate;

    @Column(name = "private_only")
    private boolean privateOnly;

    @Enumerated(EnumType.STRING)
    @Column(name = "type")
    private PaperType type;

    @ManyToMany
    @JoinTable(name = "paper_keywords",
               joinColumns = @JoinColumn(name = "paper_id"),
               inverseJoinColumns = @JoinColumn(name = "keyword_id"))
    private Set<Keyword> keywords;
}

File: mobile/types/Paper.ts
import {PaperType} from "./PaperType";

export interface Paper {
    id: number;
    title: string;
    abstractText: string;
    authors: string[];
    publishDate: string; // INFO: should probably be parsed into date
    doi: string;
    points: number;
    privateOnly: boolean;
    type: keyof typeof PaperType;
    keywords: {id: number; text: string};
}
